frontend/device/generic/device.lua

```diff
@@ -116,8 +117,14 @@ end
 -- ONLY used for Kobo and PocketBook devices
 function Device:onPowerEvent(ev)
     local Screensaver = require("ui/screensaver")
+    local network_manager = require("ui/network/manager")
     if (ev == "Power" or ev == "Suspend") and not self.screen_saver_mode then
         self.powerd:beforeSuspend()
+        self.wifi_was_on = network_manager:getWifiStatus()
```

Comment:
Depends on the status of the network, `getWifiStatus` can be blocked for 10+ seconds. So I suggest we remember the wifi status in networkmanager instead. Then just check the flag here. We can set the wifi status in `promptWifiOn`'s `ok_callback`.